src/main/java/com/google/devtools/build/lib/metrics/MetricsCollector.java

```diff
@@ -322,17 +354,62 @@ private TimingMetrics finishTimingMetrics() {
     return timingMetrics.build();
   }
 
+  /**
+   * Millisecond based duration adder which maintain initially zero long sum of microseconds.
+   * May be accessed from multiple threads.
+   * As it maintains duration as long microseconds nanoseconds will be ignored.
+   * The value can be marked invalid with {@code invalidate()}.
+   * For example if the system receives some actions with unknown duration the value is getting unknown (invalid).
+   */
+  static class DurationAdder {
+
+    final LongAdder duration;
```

Comment:
Nit: Rename to `duration_ms` to make the unit clear.

Reply:
I guess, you mean 'us', as the unit is microseconds (μs).